src/main/java/org/apache/commons/jxpath/Functions.java

```diff
@@ -43,4 +45,15 @@ public interface Functions {
      * @return Function
      */
     Function getFunction(String namespace, String name, Object[] parameters);
+
+    /**
+     * Returns a Function, if any, for the specified namespace,
+     * name and parameter types.
+     * @param namespace ns
+     * @param name function name
+     * @param parameters Object[]
+     * @param jxPathFilter  the XPath filter
+     * @return Function
+     */
+    Function getFunction(String namespace, String name, Object[] parameters, JXPathFilter jxPathFilter);
```

Comment:
This needs to be a default method to avoid breaking binary compatibility. This might be a misconfiguration in our parent POM or a bug in japicmp...

Reply:
This method has different implementations for the three classes implementing the `Functions` interface, it cannot be made final. So, in order to not break binary compatibility, I've removed this function and made the implemented `getFunction` methods use a filter based on the `jxpath.allow.class` system property. This way we don't have to explicitly pass a filter parameter to the `getFunction` method.